### PR TITLE
115 jsondoc

### PR DESCRIPTION
**Pull request by gentee**

### CODE REVIEW

File: packages/converter/converter.go [hunk 1]
```diff
@@ -569,20 +570,31 @@ func InterfaceToStr(v interface{}) string {
...
 		}
 	}
-	return str
+	return str, nil
 }
 
 // InterfaceSliceToStr converts the slice of interfaces to the slice of strings
-func InterfaceSliceToStr(i []interface{}) []string {
-	var str []string
+func InterfaceSliceToStr(i []interface{}) (str []string, err error) {
```
Comment (ghost) [inline packages/converter/converter.go]: may be return value should be "strs" instead of "str"?

File: packages/smart/funcs.go [hunk 2]
```diff
@@ -458,6 +460,18 @@ func DBInsert(sc *SmartContract, tblname string, params string, val ...interface
 	return
 }
 
+func PrepareColumns(columns string) string {
```
Comment (ghost) [inline packages/smart/funcs.go]: add a comment please. What this func does?

File: packages/smart/smart.go [hunk 3]
```diff
@@ -538,24 +543,28 @@ func (sc *SmartContract) AccessColumns(table string, columns *[]string, update b
 		return fmt.Errorf(eTableNotFound, table)
 	}
 	var cols map[string]string
-	hcolumns := make(map[string]bool)
+	hcolumns := make(map[string]colAccess)
```
Comment (ghost) [inline packages/smart/smart.go]: more semantic names please. What is hcolumns?

**Review by ghost — CHANGES_REQUESTED** [review]

File: packages/converter/converter.go (hunk 1 above)
Comment (gentee) [reply packages/converter/converter.go]: All suggestions have been realized.

**Review by potehinre — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**